File: src/main/java/com/heybys/oddments/base/annotations/UseCase.java
package com.heybys.oddments.base.annotations;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Service;
import org.springframework.validation.annotation.Validated;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Service
@Validated
public @interface UseCase {

    @AliasFor(annotation = Service.class)
    String value() default "";
}

File: src/main/kotlin/com/heybys/oddments/base/annotations/UseCase.kt
package com.heybys.oddments.base.annotations

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Service
import org.springframework.validation.annotation.Validated

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Service
@Validated
annotation class UseCase(@get:AliasFor(annotation = Service::class) val value: String = "")
